### PR TITLE
Get gcc builds with Linux working in travis and MSVC builds working in appveyor

### PR DESCRIPTION
**Pull request by mbeach**
These commits get the the travis builds using gcc on Linux passing. There are still issues with clang and Visual Studio.

These commits limit the compiler.libcxx settings built for to libstdc++11, since the boost libraries are always built with the C++11 ABI. This can be reverted at a later time to build for libstdc++ too.

### CODE REVIEW

**Comment by mbeach** [comment]
Hmm, it appears that I spoke too soon. All of the x86 builds seem to be working, but the x86_64 shared library builds seem to still have an issue. I will investigate, but this problem notwithstanding, I still think that these changes are worthwhile.

**Comment by mbeach** [comment]
Pull request bincrafters/conan-lz4#7 should get the x86_64 shared library builds working.

**Comment by mbeach** [comment]
It now builds for Windows too! Windows shared builds are disabled, as they fail in with missing symbols and it looks like a tricky one to track down. This is probably something to come back to later if anyone missed them...

**Comment by solvingj** [comment]
Sorry, really trying to get back around to this after done with boost 1.67.0.  Really appreciate you continuing to work through this one.

**Comment by mbeach** [comment]
@solvingj No problem, and no rush. I'm perfectly content to consume this from my own repo for the time being.

**Comment by SSE4** [comment]
builds fail because of `error: an attribute list cannot appear here` for `LZ4_DEPRECATED. seems to be related to the https://github.com/lz4/lz4/issues/409

**Comment by mbeach** [comment]
@SSE4 Hi, thanks for looking into it. I actually had a commit on another branch which deals with this by moving to v1.8.2 of of lz4. I've tidied this up and included it in this pull request, but there are still a few issues that stop the clang builds succeeding.

File: appveyor.yml [hunk 1]
```diff
@@ -4,15 +4,10 @@ environment:
     PYTHON: "C:\\Python27"
     PYTHON_VERSION: "2.7.8"
     PYTHON_ARCH: "32"
+    CONAN_ARCHS: x86_64
 
     matrix:
         #- MINGW_CONFIGURATIONS: '4.9@x86_64@seh@posix, 4.9@x86_64@sjlj@posix, 4.9@x86@sjlj@posix, 4.9@x86@dwarf2@posix, 6@x86_64@seh@posix, 7@x86_64@seh@posix'
-        - APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2015
```
Comment (uilianries) [inline appveyor.yml:10]: Doesn't folly supporting MSVC 12 and 14?

File: build.py [hunk 2]
```diff
@@ -9,13 +9,16 @@
...
     
-    builder = build_template_default.get_builder()
-    
+    builder = build_template_default.get_builder(pure_c=False)
+
+    # Revert this change once boost is available for other than libstdc++11.    
     modified_builds = []
     for settings, options, env_vars, build_requires, reference in builder.items:
-        options["gflags:nothreads"] = False
+        if settings.get("compiler.libcxx") == "libstdc++":
```
Comment (uilianries) [inline build.py:17]: if it doestn't support some profile, we need to raise an exception on `configure`

**Comment by SSE4** [comment]
@uilianries is this still needed?

**Comment by uilianries** [comment]
no longer required. our latest versions has all those fixes. actually we just need to fix boost ...

**Pull request closed without merging**